internal/sync/controller.go

```diff
@@ -228,3 +235,18 @@ func maps(events []models.Event) map[string]models.Event {
 	}
 	return result
 }
+
+func removeDeclinedEvents(events []models.Event) []models.Event {
+	var toDeleteIndex []int
+	for i, event := range events {
+		log.Warn("event accept check", "accepted", event.Accepted)
+		if !event.Accepted {
+			toDeleteIndex = append(toDeleteIndex, i)
+		}
+	}
+	for _, k := range toDeleteIndex {
+		log.Warn("Deleting event id", "id", k)
+		events = slices.Delete(events, k, k+1)
```

Comment:
That loop would only work when iterating backwards through `toDeleteIndex`. Currently after deleting the first event, all future indexes are off by one.

The following is much simpler (and doesn't even require Go 1.21):
```
	var filteredEvents []models.Event
	for i, event := range events {
		log.Debug("event accept check", "accepted", event.Accepted)
		if event.Accepted {
			filteredEvents = append(filteredEvents, event)
		}
	}
	return filteredEvents
```

Reply:
Thanks for the hint! :)